vmdb/app/models/storage.rb

```diff
@@ -100,6 +100,12 @@ def my_zone
     return self.ext_management_systems.first.my_zone
   end
 
+  def self.without_hosts
+    habtm_table = Arel::Table.new(:hosts_storages)
+    join_table_with_condition = habtm_table.project(habtm_table[:storage_id])
```

Comment:
Was thinking something like...

```ruby
def self.without_hosts
  includes(:hosts).where(:hosts_storages => {:storage_id => nil})
end
```

This works (I tested locally), and it's a lot easier to read.